gpkit/varkey.py

```diff
@@ -73,6 +70,9 @@ def __init__(self, name=None, **kwargs):
             self.keys.add(self.str_without(["idx"]))
             self.keys.add(self.str_without(["idx", "modelnums"]))
 
+        self.hmap = NomialMap({HashVector({self: 1}): 1.0})
+        self.hmap.units = self.units
```

Comment:
it removes a conditional in both substitution and Nomial creation, so it's pretty much always needed.